test/Java9andUp/src/org/openj9/test/java9AttachAPI/TargetManager.java

```diff
@@ -0,0 +1,339 @@
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testng.log4testng.Logger;
+
+/* This is was copied from org.openj9.test.attachAPI.TargetManager in Java8AndUp.
+ * Unnecessary code has been trimmed.
+ */
+@SuppressWarnings("nls")
+class TargetManager {
+	private static Logger logger = Logger.getLogger(TargetManager.class);
+	public static final String VMID_PREAMBLE = "vmid=";
+	public static final String STATUS_PREAMBLE = "status=";
+	public static final String STATUS_INIT_SUCESS = "init_success";
+	public static final String STATUS_INIT_FAIL = "attach_init_fail";
+
+	String targetId, displayName;
+	private Process proc;
+	private BufferedWriter targetInWriter;
+	private BufferedReader targetOutReader;
+	private BufferedReader targetErrReader;
+	private OutputStream targetIn;
+	String errOutput = "";
+	String outOutput = "";
+	private String targetVmid;
+	public static final String TARGETVM_START = "targetvm_start";
+	public static final String TARGETVM_STOP = "targetvm_stop";
+	static boolean verbose = false;
+	private static boolean doLogging = false;
+	private static final String DEFAULT_IPC_DIR = ".com_ibm_tools_attach";
+	public TargetStatus targetVmStatus;
+	private boolean active = true;
+
+	public TargetStatus getTargetVmStatus() {
+		return targetVmStatus;
+	}
+
+	public enum TargetStatus {
+		INIT_SUCCESS, INIT_FAILURE
+	}
+
+	public static void setVerbose(boolean v) {
+		verbose = v;
+	}
+
+	public BufferedWriter getTargetInWriter() {
+		return targetInWriter;
+	}
+
+	public BufferedReader getTargetOutReader() {
+		return targetOutReader;
+	}
+
+	public BufferedReader getTargetErrReader() {
+		return targetErrReader;
+	}
+
+	public TargetManager(String cmdName, String targetId) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, null, null);
+	}
+
+	/*
+	 * target must print the PID on one line, other information on following
+	 * line(s) (if any), the initialization status on the final line.
+	 */
+	public TargetStatus readTargetStatus() {
+		TargetStatus tgtStatus = TargetStatus.INIT_FAILURE;
+		StringBuffer targetLog = new StringBuffer();
+		try {
+			boolean done = false;
+			do {
+				String tgtOutput = targetOutReader.readLine();
+				if (null != tgtOutput) {
+					targetLog.append(tgtOutput);
+					targetLog.append('\n');
+				} else {
+					logger.debug("TargetVM stdout closed unexpectedly");
+					while (targetErrReader.ready()) {
+						String currentLine = targetErrReader.readLine();
+						if (null != currentLine) {
+							logger.error(currentLine + "\n");
+						}
+					}
+					tgtStatus = TargetStatus.INIT_FAILURE;
+					break;
+				}
+				if (verbose) {
+					logger.debug("TargetVM output: " + tgtOutput);
```

Comment:
Does this need to be wrapped in a `verbose` test?  Won't the logger.debug call determine whether it should be printed or not based on the configured logging level?